### ISSUE TITLE
questions

### ISSUE THREAD

**Issue by Roki100** [HermanFassett/youtube-scrape#22]
Hi! I checked ur api and i noticed that when i refresh few times i get different json sometimes so my questions are:
* do you use some sort of proxies?
* did you ever hit 429 on search pages on youtube?
* do you know if youtube has some limit of requests to search results before hitting 429?

im trying to find out how does yt ratelimiting on search work, and found your heroku api that is public and works so im asking out of curiosity

**Comment by HermanFassett** [comment]
No proxies built in.
YouTube made a recent update so I had to change the code to work with that. It appears with the new update the results aren't as consistent; also the new update isn't always in affect (I'm assuming it will be more consistent as update is rolled out) so the code uses the old method of parsing search results when update isn't used, which will have a different return. (This is denoted in json results. "parser": "json_format" means new YouTube search is used, while "html_format" is old search.)

Previously I did hit request limits in development, though I never pinned down what the definition was (someone mentioned around 150 requests but I'm pretty sure I broke that limit). In development for the new update I made what felt like a large number of requests and never hit a limit. That said, there probably is a limit, and I'm also using free heroku hosting so there is also a limited amount of "up" hours during any given month.

Deploying your own version of youtube-scrape to make requests would decrease your chances of hitting a rate/hosting limit, but if you're looking for a reliable API, I would look into the YouTube Data API to have a known and probably much higher limit.
